r/tests/testthat/test-dataset.R

```diff
@@ -397,6 +397,59 @@ test_that("Partitioning inference", {
   )
 })
 
+test_that("Specifying partitioning when hive_style", {
+  expected_schema <- open_dataset(hive_dir)$schema
+
+  # If string and names match hive partition names, it's accepted silently
+  ds_with_chr <- open_dataset(hive_dir, partitioning = c("group", "other"))
+  expect_equal(ds_with_chr$schema, expected_schema)
```

Comment:
Do we want to test that the contents of the partitioning columns _don't_ have the partitioning bits in them too? Or maybe those are tested elsewhere?

Reply:
It's covered here in that `group` wouldn't be `int32` (what we detect, which is tested elsewhere), it would be `string`. This test fails on master.

Reply:
Aaaah right, excellent